resampy/core.py

```diff
@@ -120,6 +120,106 @@ def resample(x, sr_orig, sr_new, axis=-1, filter='kaiser_best', **kwargs):
     # Construct 2d views of the data with the resampling axis on the first dimension
     x_2d = x.swapaxes(0, axis).reshape((x.shape[axis], -1))
     y_2d = y.swapaxes(0, axis).reshape((y.shape[axis], -1))
-    resample_f(x_2d, y_2d, sample_ratio, interp_win, interp_delta, precision)
+
+    scale = min(1.0, sample_ratio)
+    time_increment = 1./sample_ratio
+    t_out = np.arange(shape[axis]) * time_increment
+
+    resample_f(x_2d, y_2d, t_out, interp_win, interp_delta, precision, scale)
+
+    return y
+
+
+def resample_nu(x, t_out, axis=-1, filter='kaiser_best', **kwargs):
```

Comment:
Something seems wrong here - if `sr_orig` is not provided, how do we relate `t_out` to the sample positions in x?

Reply:
In this implementation the non-uniform sampling is only allowed on the output.
With this assumption it is possible to normalize all coordinates w.r.t. the input spacing.
The input axis is always [0, 1, 2, 3, ...], and `t_out` contains normalized coordinates.

Reply:
That makes sense.  I don't think we should put the onus on the user to normalize sample positions though - this seems like all too easy of a thing to miss unless you're reading the docstring very carefully.  Additionally, if we want to force the input to be uniformly sampled, I think it makes sense to require `sr_orig` as a parameter to provide a strong cue to the user.

I don't mind using normalized time for internal functions, where it makes a lot of sense from a computational perspective.  I'm mostly worried about API and ease-of-use here.

Reply:
OK I have just pushed a new version with the following signature:
```python
def resample_nu(x, t_out, sr_orig=1., ...) 
```
if you prefer I could change it as follow:
```python
def resample_nu(x, sr_orig, t_out, ...) 
```
to be more in line with the `resample` function.

Reply:
Thanks!  I actually do think it would be best to make the input sampling rate a required argument here, both for consistency with the existing API and as a safeguard against accidental misuse.

If we do that, then we can relax `t_out` to simply be `time` rather than normalized time, which is also a bit easier to explain.

Reply:
done